archive/imo/imo2013_q5.lean

```diff
@@ -0,0 +1,413 @@
+/-
+Copyright (c) 2020 David Renshaw. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Renshaw
+-/
+
+import algebra.geom_sum
+import data.rat.basic
+import data.rat.order
+import data.real.basic
+
+/-!
+# IMO 2013 Q5
+
+Let ℚ>₀ be the set of positive rational numbers. Let f: ℚ>₀ → ℝ be a function satisfying
+the conditions
+
+  (1) f(x) * f(y) ≥ f(x * y)
+  (2) f(x + y)    ≥ f(x) + f(y)
+
+for all x,y ∈ ℚ>₀. Given that f(a) = a for some rational a > 1, prove that f(x) = x for
+all x ∈ ℚ>₀.
+
+# Solution
+
+We provide a direct translation of the solution found in
+https://www.imo-official.org/problems/IMO2013SL.pdf
+-/
+
+open_locale big_operators
+
+lemma factor_xn_minus_yn (x : ℝ) (y : ℝ) (n : ℕ)
+      :  x^n - y^n = (x - y) * (∑ (i:ℕ) in finset.range n, (x ^(i) * y ^(n - 1 - i))) :=
```

Comment:
Here and below, we put colons at the end of lines rather than at the beginning, see our [code style guide](https://leanprover-community.github.io/contribute/style.html):
```suggestion
lemma factor_xn_minus_yn (x : ℝ) (y : ℝ) (n : ℕ) :
  x^n - y^n = (x - y) * (∑ (i:ℕ) in finset.range n, (x ^(i) * y ^(n - 1 - i))) :=
```